engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -256,20 +255,30 @@ class WorkflowActor(workflowId: WorkflowId,
 
   onTransition {
     case oldState -> terminalState if terminalState.terminal =>
-      log.info(s"$tag transition from $oldState to $terminalState: shutting down")
+      workflowLogger.info(s"transition from $oldState to $terminalState: shutting down")
       // Add the end time of the workflow in the MetadataService
       val now = OffsetDateTime.now
-      val metadataEventMsg = MetadataEvent(MetadataKey(workflowId, None, WorkflowMetadataKeys.EndTime),
-        MetadataValue(now))
+      val metadataEventMsg = MetadataEvent(MetadataKey(workflowId, None, WorkflowMetadataKeys.EndTime), MetadataValue(now))
       serviceRegistryActor ! PutMetadataAction(metadataEventMsg)
       terminalState match {
         case WorkflowSucceededState =>
           context.parent ! WorkflowSucceededResponse(workflowId)
         case WorkflowFailedState => // The failure will already have been sent. So just kick back and relax here.
         case WorkflowAbortedState =>
           context.parent ! WorkflowAbortedResponse(workflowId)
-        case unknownState => log.warning(s"$tag $unknownState is an unhandled terminal state!")
+        case unknownState => workflowLogger.warn(s"$unknownState is an unhandled terminal state!")
       }
+
+      // Copy/Delete workflow logs
+      if (WorkflowLogger.isEnabled) {
+        stateData.workflowDescriptor foreach { wd =>
+          // Create the actor in the system context because this WorkflowActor is going to die right after
+          val copyWorkflowLogsActor: ActorRef = context.system.actorOf(CopyWorkflowLogsActor.props(wd), s"CopyWorkflowLogsActor-$workflowId")
+          copyWorkflowLogsActor ! CopyWorkflowLogsActor.Start
```

Comment:
I'm not a wheeled reviewer, so feel free to ignore, but--

I am _not_ a fan of:

- Create an actor
- Send the actor a message
- Kill yourself

If the actor decides to restart, and there is no supervisor, then the actor will sit there idly waiting for someone, anyone, to tell it what to do... forever alone... 😢 

Instead, I prefer actors like the CopyWorkflowLogsActor know what their purpose in life is, via a value or message the constructor if necessary, and starting running in the `preStart()`.